applications/services/cli/cli_commands.c

```diff
@@ -396,6 +396,139 @@ void cli_command_led(Cli* cli, FuriString* args, void* context) {
     furi_record_close(RECORD_NOTIFICATION);
 }
 
+void cli_command_buzzer_usage(bool is_freq, FuriString* args) {
+    if(is_freq) {
+        cli_print_usage(
+            "buzzer freq", "<freq_hz> [<0-...>[<ms|s|m|h>]]", furi_string_get_cstr(args));
+
+    } else {
+        cli_print_usage("buzzer note", "<note> [<0-...>[<ms|s|m|h>]]", furi_string_get_cstr(args));
+    }
+}
+
+float cli_command_buzzer_frequency(bool is_freq, FuriString* args) {
+    float frequency = 0.0f;
+
+    if(is_freq) {
+        args_read_float_and_trim(args, &frequency);
+        return frequency;
+    }
+
+    // Extract note frequency
+
+    FuriString* note_string;
+    note_string = furi_string_alloc();
+
+    do {
+        if(!args_read_string_and_trim(args, note_string)) {
+            break;
+        }
+        frequency = note_frequency(furi_string_get_cstr(note_string));
+    } while(false);
+
+    furi_string_free(note_string);
+
+    return frequency;
+}
+
+void cli_command_buzzer_play(NotificationApp* notification, bool is_freq, FuriString* args) {
+    FuriString* delay_string;
+    delay_string = furi_string_alloc();
+
+    do {
+        float frequency = cli_command_buzzer_frequency(is_freq, args);
+        if(frequency <= 0.0f) {
+            cli_command_buzzer_usage(is_freq, args);
+            break;
+        }
+
+        const NotificationMessage notification_buzzer_message = {
+            .type = NotificationMessageTypeSoundOn,
+            .data.sound.frequency = frequency,
+            .data.sound.volume = 1.0,
+        };
+
+        // Optional delay
+        uint32_t delay_ms = 100;
+        if(args_read_string_and_trim(args, delay_string)) {
+            if(!args_read_time(delay_string, &delay_ms, "ms")) {
+                cli_command_buzzer_usage(is_freq, args);
+                break;
+            }
+        }
+
+        const NotificationMessage notification_delay_message = {
+            .type = NotificationMessageTypeDelay,
+            .data.delay.length = delay_ms,
+        };
+
+        const NotificationSequence notification_sequence = {
+            &notification_buzzer_message,
+            delay_ms == 0 ? &message_do_not_reset : &notification_delay_message,
+            NULL,
+        };
+
+        // Send notification
+        notification_message_block(notification, &notification_sequence);
```

Comment:
@skotopes Is there a way to interrupt notification by Ctrl+C? I would be grateful for examples

Reply:
no

Reply:
I've added function `cli_sleep` for this purpose. Please take a look